### ISSUE TITLE
Carbon v9 Number Input shows accessibility issue with DAP

### ISSUE THREAD

**Issue by SowmyaRaviKrishnan** [carbon-design-system/carbon#4655]
Carbon v9 Number Input shows below accessibility issue with DAP
1.Id references used in WAI-ARIA properties must be valid

## Environment

> Operating system- RHEL

> Browser - Chrome

> Assistive technology used to verify - IBM DAP

## Detailed description

> What version of the Carbon Design System are you using? - v9

> What WCAG 2.1 checkpoint does the issue violate?
IBM 1.3.1 Info and Relationships
IBM 4.1.2 Name, Role, Value

**Comment by SowmyaRaviKrishnan** [comment]
@joshblack  @amuqeetkhan

**Comment by abbeyhrt** [comment]
Hi @SowmyaRaviKrishnan, thanks for bringing this to our attention! As I've mentioned in the other issues for v9,we will not be backporting WCAG 2.1 fixes to v9 but we always accept community contributions for any v9 issues! If you would like this to be addressed, feel free to open up a PR and we’ll review it! 👍

**Comment by amuqeetkhan** [comment]
Hello Everyone, This is not a v9 problem. We have tried this on v10 actually as per our package.json
```
"dependencies": {
    "@carbon/colors": "^10.2.0",
    "@carbon/grid": "^10.3.0",
    "@carbon/icons": "^10.3.0",
    "@carbon/icons-react": "^10.3.0",
    "@carbon/layout": "^10.2.0",
    "@carbon/type": "^10.2.0",
    "carbon-components": "^10.3.2",
    "carbon-components-react": "^7.3.2",
    "carbon-icons": "^7.0.7",
```
So we should consider this a v10 issue. @joshblack @abbeyhrt

**Comment by tw15egan** [comment]
Which ruleset are you using to test this against? Do you have any screenshots? Running DAP on our component does not show any errors (the only error is coming from Storybook). 


<img width="1376" alt="Screen Shot 2020-01-20 at 2 10 34 PM" src="https://user-images.githubusercontent.com/11928039/72760763-b5d12980-3b8e-11ea-9556-c87be1018107.png">

**Comment by abbeyhrt** [comment]
@amuqeetkhan I would also check and see if you still get this violation with the newest version of Carbon. We are at 10.9.2 for `carbon-components` and 7.9.2 for `carbon-components-react`. We have been resolving a lot of our DAP violations so it's very possible that this violation has been addressed recently.
